File: src/main/java/com/github/oxyethylene/otelcollector/grpc/TraceServiceImpl.java
package com.github.oxyethylene.otelcollector.grpc;

import io.grpc.stub.StreamObserver;
import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceResponse;
import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpc;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class TraceServiceImpl extends TraceServiceGrpc.TraceServiceImplBase {
    private static final Logger log = LoggerFactory.getLogger(TraceServiceImpl.class);

    @Override
    public void export(ExportTraceServiceRequest request, StreamObserver<ExportTraceServiceResponse> responseObserver) {
        log.info("Received OTLP traces export");
        ExportTraceServiceResponse resp = ExportTraceServiceResponse.getDefaultInstance();
        responseObserver.onNext(resp);
        responseObserver.onCompleted();
    }
}

File: src/main/kotlin/com/github/oxyethylene/otelcollector/grpc/TraceServiceImpl.kt
package com.github.oxyethylene.otelcollector.grpc

import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest
import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceResponse
import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpcKt
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class TraceServiceImpl : TraceServiceGrpcKt.TraceServiceCoroutineImplBase() {
    override suspend fun export(request: ExportTraceServiceRequest): ExportTraceServiceResponse {
        log.info("Received OTLP traces export")
        return ExportTraceServiceResponse.getDefaultInstance()
    }

    companion object {
        private val log = LoggerFactory.getLogger(TraceServiceImpl::class.java)
    }
}
